src/components/POLine/POLineAgreementLines/POLineAgreementLinesContainer.js

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { stripesConnect } from '@folio/stripes/core';
+import {
+  Accordion,
+  Badge,
+  Loading,
+} from '@folio/stripes/components';
+import {
+  useShowCallout,
+} from '@folio/stripes-acq-components';
+
+import {
+  AGREEMENT_LINES,
+} from '../../Utils/resources';
+
+import { ACCORDION_ID } from '../const';
+import POLineAgreementLines from './POLineAgreementLines';
+
+const POLineAgreementLinesContainer = ({ lineId, label, mutator }) => {
+  const [agreementLines, setAgreementLines] = useState();
+  const showCallout = useShowCallout();
+
+  useEffect(() => {
+    setAgreementLines();
+
+    mutator.agreementLines.GET()
+      .then(setAgreementLines)
+      .catch(() => {
+        showCallout({ messageId: 'ui-orders.relatedAgreementLines.actions.load.error', type: 'error' });
+      });
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [lineId, showCallout]);
+
+  if (!agreementLines) return <Loading />;
+
+  return (
+    <Accordion
+      displayWhenClosed={<Badge>{agreementLines?.length}</Badge>}
```

Comment:
will it be empty badge when loading?

Reply:
it will show spinner, no `Accordion`

Reply:
ah I see, not the best idea to show infinite spinner when error is happened (toast doesn't solve this)

Reply:
actually, toast is kind of a solution - it shows to user that error is happened, so user can close the pane or refresh the page

Reply:
what's more confusing - show just the same component with empty values, that could make an impression like there is empty thing instead of error. From my perspective if we can't guarantee right program flow without back-end data - it's better to not show component with controls at all and spinner is not the worst idea